server/sonar-main/src/main/java/org/sonar/application/config/AppSettingsLoaderImpl.java

```diff
@@ -74,9 +74,9 @@ public AppSettings load() {
 
   private static File detectHomeDir() {
     try {
-      File appJar = new File(AppSettingsLoaderImpl.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+      File appJar = new File(Class.forName("org.sonar.application.App").getProtectionDomain().getCodeSource().getLocation().toURI());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this use of dynamic class loading. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2658)